src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -137,34 +133,48 @@ export async function promptForArguments( preselectedOptions: InstanceOptions, d
 		elasticsearch: preselectedOptions.elasticsearch || defaultOptions.elasticsearch || DEV_ENVIRONMENT_DEFAULTS.elasticsearchVersion,
 		mariadb: preselectedOptions.mariadb || defaultOptions.mariadb || DEV_ENVIRONMENT_DEFAULTS.mariadbVersion,
 		mediaRedirectDomain: preselectedOptions.mediaRedirectDomain || '',
-		wordpress: {},
-		muPlugins: {},
-		clientCode: {},
+		wordpress: {
+			mode: 'image',
+		},
+		muPlugins: {
+			mode: 'image',
+		},
+		clientCode: {
+			mode: 'image',
+		},
 		statsd: false,
 		phpmyadmin: false,
 		xdebug: false,
+		siteSlug: '',
 	};
 
 	if ( ! instanceData.mediaRedirectDomain && defaultOptions.mediaRedirectDomain ) {
 		const mediaRedirectPromptText = `Would you like to redirect to ${ defaultOptions.mediaRedirectDomain } for missing media files?`;
 		const setMediaRedirectDomain = await promptForBoolean( mediaRedirectPromptText, true );
 		if ( setMediaRedirectDomain ) {
-			instanceData.mediaRedirectDomain = defaultOptions.mediaRedirectDomain;
+			instanceData.mediaRedirectDomain = defaultOptions.mediaRedirectDomain ?? '';
 		}
 	}
 
 	for ( const component of DEV_ENVIRONMENT_COMPONENTS ) {
-		const option = preselectedOptions[ component ];
-		const defaultValue = defaultOptions[ component ];
+		const option = preselectedOptions[ component ].toString();
+		const defaultValue = ( defaultOptions[ component ] ?? '' ).toString();
 
-		instanceData[ component ] = await processComponent( component, option, defaultValue );
+		const result = await processComponent( component, option, defaultValue );
+		if ( null === result ) {
+			throw new Error( 'processComponent() returned null' );
+		}
```

Comment:
Now, `processComponent()` is allowed to return `null`, and `null` is not allowed here. I am not sure that to throw an exception is the best way here, but I don't know what else we can do.